python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -40,19 +40,68 @@
 
 
 class ResourceDemandScheduler:
-    def __init__(self, provider: NodeProvider,
+    def __init__(self,
+                 provider: NodeProvider,
                  node_types: Dict[NodeType, NodeTypeConfigDict],
-                 max_workers: int):
+                 max_workers: int,
+                 aggressive: bool = False,
+                 target_utilization_fraction: float = 1):
         self.provider = provider
         self.node_types = copy.deepcopy(node_types)
         self.max_workers = max_workers
-        # is_legacy_yaml tracks if the cluster configs was originally without
-        # available_node_types and was autofilled with available_node_types.
-        self.is_legacy_yaml = (NODE_TYPE_LEGACY_HEAD in node_types
-                               and NODE_TYPE_LEGACY_WORKER in node_types)
+        self.aggressive = aggressive
+        self.target_utilization_fraction = target_utilization_fraction
```

Comment:
Can we call self.reset_config() at the end of the constructor to ensure the code path is exercised?

Reply:
Then outside we will call:
self.resource_demand_scheduler.__init__(..)?
And the self.node_types will be set in the reset_config ? (We won't set it before)

Reply:
No, you'd always call reset config. You can move the init logic to the reset config call.

Reply:
I am a bit confused. The `reset_config` is called on a ResourceDemandScheduler instance that was already instantiated.
```
            if hasattr(self, "resource_demand_scheduler"):
                # The node types are autofilled internally for legacy yamls,
                # overwriting the class will remove the inferred node resources
                # for legacy yamls.
                self.resource_demand_scheduler.reset_config(
                    self.provider, self.available_node_types,
                    self.config["max_workers"],
                    self.config["autoscaling_mode"] == "aggressive",
                    self.config["target_utilization_fraction"])
            else:
                self.resource_demand_scheduler = ResourceDemandScheduler(
                    self.provider, self.available_node_types,
                    self.config["max_workers"],
                    self.config["autoscaling_mode"] == "aggressive",
                    self.config["target_utilization_fraction"])
```